packages/next/next-server/lib/router/router.ts

```diff
@@ -17,10 +17,15 @@ import { isDynamicRoute } from './utils/is-dynamic'
 import { getRouteMatcher } from './utils/route-matcher'
 import { getRouteRegex } from './utils/route-regex'
 
-function addBasePath(path: string): string {
-  // @ts-ignore variable is always a string
-  const p: string = process.env.__NEXT_ROUTER_BASEPATH
-  return path.indexOf(p) !== 0 ? p + path : path
+// @ts-ignore variable is always a string
+const basePath: string = process.env.__NEXT_ROUTER_BASEPATH
+
+export function addBasePath(path: string): string {
+  return path.indexOf(basePath) !== 0 ? basePath + path : path
+}
+
+function delBasePath(path: string): string {
+  return path.replace(new RegExp(`^${basePath}`), '')
```

Comment:
Maybe regex is a bit overkill? I think the approach of `addBasePath` would be more performant
```suggestion
  return path.indexOf(basePath) === 0 ? path.slice(basePath.length) : path
```

Reply:
Updated to not use `regex`